src/Tools/dotnet-monitor/CollectionRules/Actions/CollectLogsAction.cs

```diff
@@ -2,34 +2,108 @@
         {
-            return new CollectLogsAction();
+            if (null == options)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            ValidationContext context = new(options, _serviceProvider, items: null);
+            Validator.ValidateObject(options, context, validateAllProperties: true);
+
+            return new CollectLogsAction(_serviceProvider, endpointInfo, options);
         }
 
         private sealed class CollectLogsAction :
-            ICollectionRuleAction
+            CollectionRuleActionBase<CollectLogsOptions>
         {
-            public Task StartAsync(CancellationToken token)
+            private readonly IServiceProvider _serviceProvider;
+
+            public CollectLogsAction(IServiceProvider serviceProvider, IEndpointInfo endpointInfo, CollectLogsOptions options)
+                : base(endpointInfo, options)
             {
-                throw new NotImplementedException("TODO: Implement");
+                _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
             }
 
-            public Task<CollectionRuleActionResult> WaitForCompletionAsync(CancellationToken token)
+            protected override async Task<CollectionRuleActionResult> ExecuteCoreAsync(
+                TaskCompletionSource<object> startCompletionSource,
+                CancellationToken token)
             {
-                throw new NotImplementedException("TODO: Implement");
+                TimeSpan duration = Options.Duration.GetValueOrDefault(TimeSpan.Parse(CollectLogsOptionsDefaults.Duration));
+                bool useAppFilters = Options.UseAppFilters.GetValueOrDefault(CollectLogsOptionsDefaults.UseAppFilters);
+                LogLevel defaultLevel = Options.DefaultLevel.GetValueOrDefault(CollectLogsOptionsDefaults.DefaultLevel);
+                Dictionary<string, LogLevel?> filterSpecs = Options.FilterSpecs;
+                string egressProvider = Options.Egress;
+                LogFormat logFormat = Options.Format.GetValueOrDefault(CollectLogsOptionsDefaults.Format);
+
+                var settings = new EventLogsPipelineSettings()
+                {
+                    Duration = duration,
+                    LogLevel = defaultLevel,
+                    UseAppFilters = useAppFilters,
+                    FilterSpecs = filterSpecs
+                };
+
+                string logsFilePath = await StartLogs(startCompletionSource, EndpointInfo, settings, egressProvider, logFormat, token);
+
+                return new CollectionRuleActionResult()
+                {
+                    OutputValues = new Dictionary<string, string>(StringComparer.Ordinal)
+                    {
+                        { CollectionRuleActionConstants.EgressPathOutputValueName, logsFilePath }
+                    }
+                };
+            }
+
+            private async Task<string> StartLogs(
```

Comment:
There isn't much value in separating this out from the ExecuteCoreAsync method; please inline it.